### PR TITLE
Block on mobile unit tests fail and instructions on how to debug

### PR DESCRIPTION
**Pull request by hypest**
## Description
This PR makes the native mobile tests Travis job necessary to succeed for the whole Travis run to "green".

## How has this been tested?
Online via Travis itself on this PR and locally by running `npm run test-unit:native:debug`.

## Types of changes
1. Lift the "allow to fail" flagging in the Travis config for the native mobile tests job
2. Add documentation on how to debug the native mobile unit tests
3. Fix the `test-unit:native:debug` npm script to indeed wait for the node debugger to attach

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by gziolo** [comment]
Works for me, but I'm not the best person to validate changes, @Tug maybe? :)

**Review by Tug — APPROVED** [review]
Yep the `--inspect-brk` option should do the trick 👍
Nice addition to the documentation!

File: test/native/jest.config.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ module.exports = {
 	],
 	testURL: 'http://localhost/',
 	// Add the `Libraries/Utilities` subfolder to the module directories, otherwise haste/jest doesn't find Platform.js on Travis
-	moduleDirectories: [ 'node_modules', 'node_modules/react-native/Libraries/Utilities' ],
+	// And, add it first so https://github.com/facebook/react-native/blob/v0.60.0/Libraries/react-native/react-native-implementation.js#L324-L326 doesn't pick up the Platform npm module.
```
Comment (gziolo) [inline test/native/jest.config.js]: Super nit
```suggestion
	// and, add it first so https://github.com/facebook/react-native/blob/v0.60.0/Libraries/react-native/react-native-implementation.js#L324-L326
	// doesn't pick up the Platform npm module.
```

**Comment by gziolo** [comment]
There is also the Testing Overview document:
https://github.com/WordPress/gutenberg/blob/master/docs/contributors/testing-overview.md

We should either move the details about testing there and cross-reference or the other way around. This is really important so it's easy to find details about the native app.

File: test/native/jest.config.js (hunk 1 above)
Comment (hypest) [reply test/native/jest.config.js]: Good catch, fixed with 91c96b7.

**Comment by hypest** [comment]
> There is also the Testing Overview document:
https://github.com/WordPress/gutenberg/blob/master/docs/contributors/testing-overview.md
>
> We should either move the details about testing there and cross-reference or the other way around. This is really important so it's easy to find details about the native app.

Makes sense and I was thinking something along those line too 👍. Addressed with e5a0a97.

File: docs/contributors/testing-overview.md [hunk 2]
```diff
@@ -386,3 +386,18 @@ To run unit tests only, without the linter, use `npm run test-unit-php` instead.
 
 [snapshot testing]: https://jestjs.io/docs/en/snapshot-testing.html
 [update snapshots]: https://jestjs.io/docs/en/snapshot-testing.html#updating-snapshots
+
+## Native mobile testing
```
Comment (gziolo) [inline docs/contributors/testing-overview.md]: Can you move it up just before the section about E2E testing? It's better when it's grouped together with the regular unit testing, it's nearly identical after all :)

**Review by gziolo — APPROVED** [review]
Thank you for working for all those additions to contributors guide ❤️

File: docs/contributors/testing-overview.md (hunk 2 above)
Comment (hypest) [reply docs/contributors/testing-overview.md]: Sure, addressed with 3064350.

**Comment by gziolo** [comment]
Nice, thanks, feel free to merge 👍

**Comment by hypest** [comment]
I'm going to leave this PR open for another day before merging. We brought this up more widely in the #core-editor weekly chat yesterday so, will leave some time for people to comment if any.

**Comment by hypest** [comment]
No new comments here so, will go ahead and merge.

**Pull request merged**